content/docs/atom-guide/dev-guide/shaders/azsl/shader-variant-options.md

```diff
@@ -1,45 +1,51 @@
+linktitle: Shader variant options
+title: Shader variant options in Amazon Shading Language (AZSL)
+description: Learn about Amazon Shading Language (AZSL) shader variant options for Atom Renderer. 
 weight: 100
 ---
 
-Shader Variant Options are "statically optimizable" Shader Constants that the developer can choose to compile as static constants or as regular global variables. They are only used in conditional statements.  
+Shader variant options are shader constants that are only used in conditional statements that can be statically optimized. You can choose to compile them as static constants or as global variables.  
   
-Shader Variant Options are conceptually equivalent to [Specialization Constants](https://www.khronos.org/registry/vulkan/specs/1.1-khr-extensions/html/chap10.html#pipelines-specialization-constants) (external link) in Vulkan.  
+Shader variant options are conceptually equivalent to [Specialization Constants](https://www.khronos.org/registry/vulkan/specs/1.1-khr-extensions/html/chap10.html#pipelines-specialization-constants) (external link) in Vulkan.  
   
 The idea is easier to explain with some code. Consider the following AZSL code snippet, where a Shader Variant Option is being used to conditionally branch:  
-  
+```cpp
     if (o_useRed) {
         color = float3(1, 0, 0);
     } else {
         color = float3(0, 0, 1);
     }
-  
-When the shader code, shown above, is compiled and the value of `o_useRed` is not known at compilation time then `o_useRed` will be compiled as a regular Shader Constant. The shader byte code **will branch at runtime** depending on the value of `o_useRed` as defined by the Application.  
-  
-Please notice that **"will branch at runtime"** was emphasized. In general, branching is slower than not branching. To avoid branching, the value of `o_useRed` can be defined at compilation time, let's say to `true`.  
-  
+```
+In the shader code above, the value of `o_useRed` is not known at compilation time, so `o_useRed` is compiled as a shader constant. The shader byte code branches at runtime depending on the runtime value of `o_useRed`.  In general, branching is slow and should be avoided.
+In this scenario, if the value of `o_useRed` is defined as `true` at compilation time, then the compiler produces the following branch-less code.  
+```cpp
     // Because the developer chose o_useRed to be true at compilation time,
     // the compiler will optimize and produce the following branch-less code.
     color = float3(1, 0, 0);
-  
-Alternatively, at compilation time, `o_useRed` may be defined as `false`, and the optimized output code will look like:  
-  
+```
+If `o_useRed` is defined as `false` at compilation time, then the compiler produces the following branch-less code:  
+```cpp
     // Because the developer chose o_useRed to be false at compilation time,
     // the compiler will optimize and produce the following branch-less code.
```

Comment:
Remove the comments below (`// ...`). You already explain this in the sentence above. 
```suggestion
```

Reply:
Removed.